### PR TITLE
Replacing traceback with error message on empty query (try_usage)

### PR DESCRIPTION
**Pull request by mUtterberg**
Addresses issue  #15 

I didn't think to run tox before making changes, so I'm not 100% sure the `ERROR: InvocationError for command '/active-data-recipes/.tox/flake8/bin/flake8 adr' (exited with code 1)` was introduced by my change. However, it does pass all other tests.

Changed try_usage.py and cli.py only. Could use refactor as time allows, but currently returns the following error message rather than the previous traceback:

`Error: The queried data does not exist. (ActiveData didn\'t return any data.)`

### CODE REVIEW

**Comment by ahal** [comment]
No worries :)

You can test it out locally by running:

    $ pip install tox
    $ tox

If that passes, you can be pretty confident that the TravisCI will also pass. You could also just run flake8 directly to avoid the unittests if you want.

I will ask that you squash your commits though so it's easier for me to review them. A common way of doing this is with `git rebase -i <base commit>`. E.g:

    $ git rebase -i master

(you'll need to use -f to push afterwards, but that's fine when you're pushing to a PR on your local fork)

Thanks for the PR!

**Comment by mUtterberg** [comment]
Totally makes sense! Will do.

**Comment by mUtterberg** [comment]
Squashed! Yeah, that looks way cleaner.

File: adr/cli.py [hunk 1]
```diff
@@ -100,7 +100,9 @@ def run_recipe(recipe, args, fmt='table'):
     mod = importlib.import_module(modname, package='adr')
     output = mod.run(args)
 
-    if isinstance(fmt, string_types):
+    if output == 'adr_error_msg':
+        return 'Error: the queried data does not exist. (ActiveData didn\'t return any data.)'
```
Comment (ahal) [inline adr/cli.py]: nit: You can just make this message "ActiveData didn't return any data for the specified query"

**Comment by mUtterberg** [comment]
Gotcha! I think squashing this most recent change made it so that I can't comment directly on your review, but I pushed that change too soon. One more revision should do it, hopefully. X-/

**Comment by mUtterberg** [comment]
Ok THIS was what I meant to push.

File: adr/cli.py (hunk 1 above)
Comment (mUtterberg) [reply adr/cli.py]: Oh hey! I figured out where this "missing" comment box went...

File: adr/recipes/try_usage.py [hunk 2]
```diff
@@ -25,7 +26,10 @@ def run(args):
     data = next(run_query('try_commit_messages', **query_args))['data']
 
     count = defaultdict(int)
-    count['total'] = len(data['message'])
+    try:
+        count['total'] = len(data['message'])
+    except KeyError:
+        return MissingDataError
```
Comment (ahal) [inline adr/recipes/try_usage.py]: Close, you'll want to reraise this exception. You can do this with the `raise` keyword. Here's an old (but still good) article you can read if you want to learn more about re-raising exceptions:
http://www.ianbicking.org/blog/2007/09/12/re-raising-exceptions/index.html

But the main benefit of re-raising the exception (as opposed to just letting the original KeyError get raised), is that we can now be sure that this error was caused by ActiveData returning nothing, and not some other unrelated KeyError that happened to be raised elsewhere in the script.

File: adr/cli.py [hunk 3]
```diff
@@ -100,7 +101,9 @@ def run_recipe(recipe, args, fmt='table'):
     mod = importlib.import_module(modname, package='adr')
     output = mod.run(args)
 
-    if isinstance(fmt, string_types):
+    if output == MissingDataError:
+        return 'ActiveData didn\'t return any data.'
```
Comment (ahal) [inline adr/cli.py]: After fixing the other comment, you can put a `try/except` block around the call to `mod.run`. You can catch the custom exception you created and then print the error message in the `except` block. Hope that makes sense, and feel free to clarify!

Comment (mUtterberg) [reply adr/cli.py]: Sounds good! Raising an appropriate exception makes sense. I had to step away for a few days, so now there are conflicts between my PR and other updates in the meantime. Should I re-fork and resubmit or is it an easy resolution? I've not run into merge conflicts before, but the changes were simple enough that I can totally repeat if necessary.

Comment (ahal) [reply adr/cli.py]: No problem for stepping away, thanks for sticking with it! Let me try and resolve the conflicts for you, then you can continue working here.

**Comment by ahal** [comment]
I fixed the conflicts, but the function you were modifying in `cli.py` (`run_recipe`) got moved over to `recipe.py`. Github's conflict resolution UI didn't allow me to modify that file, so the changes you made in there got lost. But I think that's ok as that was the part you needed to change anyway :). Also looks like my commit broke flake8, my apologies! Shouldn't be hard to sort out though.

**Comment by mUtterberg** [comment]
No worries! Flake8 tests are pretty much the easiest to resolve.

**Comment by ahal** [comment]
Thanks! Are you interested in working on anything else? If so feel free to comment in any issues that look interesting to you.

**Pull request merged**